### A Single Neuron

Let’s say we have a single neuron, and there are three inputs to this neuron. As in most cases, when you initialize parameters in neural networks, our network will have weights initialized randomly, and biases set as zero to start. 

In [1]:
inputs = [1, 2, 3]

Each input also needs a weight associated with it. Inputs are the data that we pass into the model to get desired outputs, while the weights are the parameters that we’ll tune later on to get these results. 

In [2]:
weights = [0.2, 0.8, -0.5]

Next, we need the bias. At the moment, we’re modeling a single neuron with three inputs. The bias is an additional tunable value but is not associated with any input in contrast to the weights.

In [3]:
bias = 2

The values for weights and biases are what get “trained,” and they are what make a model actually work (or not work). These are what we need to store later.

This neuron sums each input multiplied by that input’s weight, then adds the bias. Our output would be calculated up to this point like: 

In [4]:
output = (inputs[0]*weights[0] + 
          inputs[1]*weights[1] + 
          inputs[2]*weights[2] + 
          bias)

print(output)

2.3


The output here should be **2.3**.

![Output for a single neuron](./img/diagram%201.png)